File: mobile/lib/models/appointment.dart
class Appointment {
  final String id;
  final String shopId;
  final int tokenNumber;
  final DateTime slotStart;
  final DateTime slotEnd;
  final String status;
  final bool isWalkIn;
  final Map<String, dynamic>? shopInfo;
  final Map<String, dynamic>? userInfo;

  const Appointment({
    required this.id,
    required this.shopId,
    required this.tokenNumber,
    required this.slotStart,
    required this.slotEnd,
    required this.status,
    this.isWalkIn = false,
    this.shopInfo,
    this.userInfo,
  });

  factory Appointment.fromJson(Map<String, dynamic> json) {
    final shopId = json['shopId'];
    return Appointment(
      id: (json['_id'] ?? json['id']) as String,
      shopId: shopId is Map ? shopId['_id'] as String : shopId as String,
      tokenNumber: json['tokenNumber'] as int? ?? 0,
      slotStart: DateTime.parse(json['slotStart'] as String),
      slotEnd: DateTime.parse(json['slotEnd'] as String),
      status: json['status'] as String,
      isWalkIn: json['isWalkIn'] as bool? ?? false,
      shopInfo: shopId is Map ? shopId as Map<String, dynamic> : null,
      userInfo: json['userId'] is Map ? json['userId'] as Map<String, dynamic> : null,
    );
  }

  Map<String, dynamic> toJson() => {
        '_id': id,
        'shopId': shopId,
        'tokenNumber': tokenNumber,
        'slotStart': slotStart.toIso8601String(),
        'slotEnd': slotEnd.toIso8601String(),
        'status': status,
        'isWalkIn': isWalkIn,
      };
}

File: backend/src/models/Appointment.js
const mongoose = require('mongoose');

const appointmentSchema = new mongoose.Schema({
  shopId: { type: mongoose.Schema.Types.ObjectId, ref: 'Shop', required: true },
  userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', default: null },
  tokenNumber: { type: Number, required: true },
  slotStart: { type: Date, required: true },
  slotEnd: { type: Date, required: true },
  status: {
    type: String,
    enum: ['Pending', 'Completed', 'No-Show', 'Cancelled'],
    default: 'Pending',
  },
  isWalkIn: { type: Boolean, default: false },
  reminderSent: { type: Boolean, default: false },
  expireAt: { type: Date, required: true },
  createdAt: { type: Date, default: Date.now },
});

appointmentSchema.index({ expireAt: 1 }, { expireAfterSeconds: 0 });
appointmentSchema.index({ shopId: 1, status: 1, slotStart: 1 });
appointmentSchema.index({ userId: 1, status: 1 });

module.exports = mongoose.model('Appointment', appointmentSchema);
